### ISSUE TITLE
ThreadLocal in ObjectStoreLocationProvider isn't serializable

### ISSUE THREAD

**Issue by nastra** [apache/iceberg#7345]
### Apache Iceberg version

main (development)

### Query engine

None

### Please describe the bug 🐞

It appears that this has been introduced as part of #7128 
```
java.io.NotSerializableException: java.lang.ThreadLocal$SuppliedThreadLocal
Serialization stack:
	- object not serializable (class: java.lang.ThreadLocal$SuppliedThreadLocal, value: java.lang.ThreadLocal$SuppliedThreadLocal@b11d2ad)
	- field (class: org.apache.iceberg.LocationProviders$ObjectStoreLocationProvider, name: temp, type: class java.lang.ThreadLocal)
	- object (class org.apache.iceberg.LocationProviders$ObjectStoreLocationProvider, org.apache.iceberg.LocationProviders$ObjectStoreLocationProvider@7c47ae7d)
	- field (class: org.apache.iceberg.SerializableTable, name: locationProvider, type: interface org.apache.iceberg.io.LocationProvider)
	- object (class org.apache.iceberg.spark.source.SerializableTableWithSize, org.apache.iceberg.spark.source.SerializableTableWithSize@706b68e6)
```

@singhpk234 could you take a look please?

**Comment by singhpk234** [comment]
Thanks @nastra for reporting this, apologies couldn't catch this earlier. Have added a fix for it as well test's which could catch this going forward, as by default ObjectStoreLocationProvider is not used by iceberg. Please take a pass when you have some cycles :) !

**Comment by singhpk234** [comment]
can be marked as fixed.

**Comment by nastra** [comment]
fixed by #7353
